students/barecki_andrzej/lesson_03_functions/L3_Exponentiation.py

```diff
@@ -0,0 +1,46 @@
+"""
+Given a positive real number a and a non-negative integer n. Calculate an without using loops, ** operator
+or the built in function math.pow(). Instead, use recursion and the relation an=a⋅an−1. Print the result.
+"""
+import os
+
+
+def input_validation(param_type):
+    global val
+    if param_type == 'a':
+        print('Set a positive real number(a):')
+        try:
+            val = float(input())
+            if val <= 0:
+                print('Error: Value is less/equal zero.')
+                input_validation('a')
+        except ValueError:
+            print('ValueError: Incorrect set of positive real number!')
+            input_validation('a')
```

Comment:
Raczej bym tu widział pętlę zamiast rekurencji, byłoby to wydajniejsze i bardziej czytelne rozwiązanie. Nie musisz poprawiać, ale przejrzyj też pozostałe zadania gdzie zrobiłeś podobnie.